lib/buildkite/test_collector/test_links_plugin/reporter.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Buildkite::TestCollector::TestLinksPlugin
+  class Reporter
+    RSpec::Core::Formatters.register self, :example_failed, :dump_summary
+
+    def initialize(output)
+      Buildkite::TestCollector.session = Buildkite::TestCollector::Session.new
+      @output = output
+      @failed_examples = []
+    end
+
+    def example_failed(notification)
+      example = notification.example
+
+      @failed_examples << {
+        scope: example.example_group.metadata[:full_description],
+        name: example.description
+      }
+    end
+
+    def dump_summary(_notification)
```

Comment:
I think you could hook into `dump_failures` instead of `dump_summary` and then avoid some of the checks you're currently doing? 

https://rspec.info/documentation/3.7/rspec-core/RSpec/Core/Formatters/Protocol.html#dump_failures-instance_method